GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Storage.v1.Data;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    /// <summary>
+    /// Wrapper around <see cref="StorageService"/> to provide simpler operations.
+    /// </summary>
+    public class StorageClient
+    {
+        /// <summary>
+        /// The underlying storage service object used by this client.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="StorageClient"/> class only provides convenience operations built on top of
+        /// an existing service object. Any more complex operations which are not supported by this wrapper may wish
+        /// to use the same service object as the wrapper, in order to take advantage of its configuration (for authentication,
+        /// application naming etc).
+        /// </remarks>
+        public StorageService Service { get; }
+
+        /// <summary>
+        /// Constructs a new client wrapping the given <see cref="StorageService"/>.
+        /// </summary>
+        /// <param name="service">The service to wrap. Must not be null.</param>
+        public StorageClient(StorageService service)
+        {
+            Service = service;
+        }
+
+        /// <summary>
+        /// Constructs a new client by creating a <see cref="StorageService"/> for the given <see cref="BaseClientService.Initializer"/>.
+        /// </summary>
+        /// <param name="initializer">The initializer to use in the service. Must not be null.</param>
+        public StorageClient(BaseClientService.Initializer initializer) : this(new StorageService(initializer))
```

Comment:
Right. So the differences are that:

- The constructors here don't do any real work, whereas the static factory method does
- The constructors are synchronous, but the static factory method is asynchronous so *has* to be a method
- The meaning of the constructors is obvious from their parameters

I think it's definitely worth trying to come up with some style consistency on when to use constructors vs when to use factory methods - will see whether I can put something together.